xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -113,3 +116,83 @@ async def send_reliable_submission(
     return await _wait_for_final_transaction_outcome(
         transaction_hash, client, prelim_result, 0
     )
+
+
+def _is_signed(transaction: Transaction) -> bool:
+    """
+    Checks if a transaction has been signed.
+
+    Args:
+        transaction: the transaction to check.
+
+    Returns:
+        Whether the transaction has been signed
+    """
+    return (
+        transaction.signing_pub_key is not None and len(transaction.signing_pub_key) > 0
+    ) or (transaction.txn_signature is not None and len(transaction.txn_signature) > 0)
+
+
+async def _get_signed_tx(
+    transaction: Transaction,
+    wallet: Wallet,
+    client: Client,
+    check_fee: bool = True,
+    autofill: bool = True,
+) -> Transaction:
+    """
+    Initializes a signed transaction for a submit request.
+
+    Args:
+        transaction: the transaction to be submitted.
+        wallet: the wallet with which to sign the transaction.
+        client: the network client with which to submit the transaction.
+        check_fee: an optional bolean indicating whether to check if the fee is
+            higher than the expected transaction type fee. Defaults to True.
+        autofill: an optional boolean indicating whether to autofill the
+            transaction. Defaults to True.
+
+    Returns:
+        The signed transaction.
+    """
+    if _is_signed(transaction):
+        return transaction
+
+    if check_fee:
+        await _check_fee(transaction, client)
+
+    if autofill:
+        transaction = await _autofill(transaction, client)
+    return await sign(transaction, wallet, False)
+
+
+async def submit_and_wait(
+    transaction: Transaction,
```

Comment:
If a transaction is signed, it'll be a string, so this type should be `Union[Transaction, str]`. (Signed transactions are usually in binary string format, aka the "tx_blob")